census.go

```diff
@@ -432,6 +434,147 @@ func (v *vocdoniHandler) censusQueueInfo(msg *apirest.APIdata, ctx *httprouter.H
+		}
+	}
+	return v.censusToken(req.Tokens, ctx)
+}
+
+func (v *vocdoniHandler) censusTokenERC20(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	req := struct {
+		Token Token `json:"token"`
+	}{}
+	if err := json.Unmarshal(msg.Data, &req); err != nil {
+		return err
+	}
+
+	// check valid token
+	if _, err := v.airstack.BlockchainToTokenBlockchain(req.Token.Blockchain); err != nil {
+		return fmt.Errorf("invalid blockchain for token %s provided", req.Token.Address)
+	}
+	tks := make([]*Token, 1)
+	tks[0] = &req.Token
+	return v.censusToken(tks, ctx)
+}
+
+func (v *vocdoniHandler) censusToken(tokens []*Token, ctx *httprouter.HTTPContext) error {
+	censusID, err := v.cli.NewCensus(api.CensusTypeWeighted)
+	if err != nil {
+		return err
+	}
+	v.censusCreationMap.Store(censusID.String(), CensusInfo{})
+	totalAddresses := uint32(0)
+	go func() {
+		startTime := time.Now()
+		log.Debugw("building NFT based census", "censusID", censusID)
+		// get holders for each token
+		holders := make([][]string, 0)
+		v.trackStepProgress(censusID, 1, 3, func(progress chan int) {
+			holders, err = v.GetTokenHolders(tokens, censusID, progress)
+		})
+		if err != nil {
+			log.Warnw("failed to build census from NFT, cannot get holders", "err", err.Error())
+			v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: err.Error()})
+			return
+		}
+		// create census from token holders
+		var participants []*FarcasterParticipant
+		v.trackStepProgress(censusID, 2, 3, func(progress chan int) {
+			participants, totalAddresses, err = v.processCensusRecords(holders, progress)
+		})
+		if err != nil {
+			log.Warnw("failed to build census from NFT", "err", err.Error())
+			v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: err.Error()})
+			return
+		}
+		var ci *CensusInfo
+		v.trackStepProgress(censusID, 3, 3, func(progress chan int) {
+			ci, err = CreateCensus(v.cli, participants, FrameCensusTypeToken, progress)
+		})
+		if err != nil {
+			log.Errorw(err, "failed to create census")
+			v.censusCreationMap.Store(censusID.String(), CensusInfo{Error: err.Error()})
+			return
+		}
+
+		// since each participant can have multiple signers, we need to get the unique usernames
+		uniqueParticipantsMap := make(map[string]struct{})
+		for _, p := range participants {
+			uniqueParticipantsMap[p.Username] = struct{}{}
+		}
+		uniqueParticipants := []string{}
+		for k := range uniqueParticipantsMap {
+			uniqueParticipants = append(uniqueParticipants, k)
+		}
+		ci.Usernames = uniqueParticipants
+		log.Infow("census created from NFT",
+			"censusID", censusID.String(),
+			"size", len(ci.Usernames),
+			"duration", time.Since(startTime),
+			"totalAddresses", totalAddresses)
+		v.censusCreationMap.Store(censusID.String(), *ci)
+	}()
+	data, err := json.Marshal(map[string]string{"censusId": censusID.String()})
+	if err != nil {
+		return err
+	}
+	return ctx.Send(data, http.StatusOK)
+}
+
+func (v *vocdoniHandler) GetTokenHolders(
```

Comment:
In Go Getters, do not have "Get" prefix, usually. 

Also I think this method should be called something such as "tokenHoldersFromAirstack()".

Comment on the method is missing.

It should not be exported.